From cf8a12afa33ddad3d46b7e40c712358f7a93db35 Mon Sep 17 00:00:00 2001
From: Szabolcs Nagy <szabolcs.nagy@arm.com>
Date: Wed, 21 Feb 2018 15:51:22 +0000
Subject: [PATCH 10/10] aarch64: Fix bits/utmp.h for GCC 8

The same suppression of -Wstringop-truncation warnings as in commit
7532837d7b03b3ca5b9a63d77a5bd81dd23f3d9c but for aarch64.

2018-02-21  Szabolcs Nagy  <szabolcs.nagy@arm.com>

	* sysdeps/unix/sysv/linux/aarch64/bits/utmp.h (struct utmp): Use
	__attribute_nonstring__.
---
 sysdeps/unix/sysv/linux/aarch64/bits/utmp.h | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/sysdeps/unix/sysv/linux/aarch64/bits/utmp.h b/sysdeps/unix/sysv/linux/aarch64/bits/utmp.h
index 850517573e..027dceb82c 100644
--- a/sysdeps/unix/sysv/linux/aarch64/bits/utmp.h
+++ b/sysdeps/unix/sysv/linux/aarch64/bits/utmp.h
@@ -61,10 +61,13 @@ struct utmp
 {
   short int ut_type;		/* Type of login.  */
   pid_t ut_pid;			/* Process ID of login process.  */
-  char ut_line[UT_LINESIZE];	/* Devicename.  */
+  char ut_line[UT_LINESIZE]
+     __attribute_nonstring__;	/* Devicename.  */
   char ut_id[4];		/* Inittab ID.  */
-  char ut_user[UT_NAMESIZE];	/* Username.  */
-  char ut_host[UT_HOSTSIZE];	/* Hostname for remote login.  */
+  char ut_user[UT_NAMESIZE]
+     __attribute_nonstring__;	/* Username.  */
+  char ut_host[UT_HOSTSIZE]
+     __attribute_nonstring__;	/* Hostname for remote login.  */
   struct exit_status ut_exit;	/* Exit status of a process marked
 				   as DEAD_PROCESS.  */
 /* The ut_session and ut_tv fields must be the same size when compiled
-- 
2.11.1

